geo/src/algorithm/voronoi_diagram.rs

```diff
@@ -0,0 +1,970 @@
+use std::{fmt, result};
+
+use crate::{coord, Coord, GeoFloat, Line, MultiPoint, Point, Polygon, Rect, Triangle};
+
+use crate::triangulate_delaunay::{
+    DelaunayTriangle, DelaunayTriangulationError, DEFAULT_SUPER_TRIANGLE_EXPANSION,
+};
+use crate::{BoundingRect, TriangulateDelaunay};
+
+type Result<T> = result::Result<T, VoronoiDiagramError>;
+
+#[derive(Debug, Clone)]
+pub struct VoronoiComponents<T: GeoFloat> {
+    pub delaunay_triangles: Vec<Triangle<T>>,
+    pub vertices: Vec<Coord<T>>,
+    pub lines: Vec<Line<T>>,
+    pub neighbours: Vec<(Option<usize>, Option<usize>)>,
+}
+
+pub trait VoronoiDiagram<T: GeoFloat>
+where
+    f64: From<T>,
+{
+    fn compute_voronoi_components(
+        &self,
+        clipping_mask: Option<&Polygon<T>>,
```

Comment:
need to describe how this behaves when this is `None`

Reply:
I'm going to add the docs as part of the prep for upstream work.